sync/task_pool.go

```diff
@@ -127,10 +131,25 @@ func (p *TaskPool) start() {
 		p.wg.Add(1)
 		workerID := i
 		q := p.qArray[workerID%p.tQNumber]
-		go p.run(int(workerID), q)
+		p.safeRun(workerID, q)
 	}
 }
 
+func (p *TaskPool) safeRun(workerID int, q chan task) {
+	gxruntime.GoSafely(nil, false,
+		func() {
+			err := p.run(int(workerID), q)
+			if err != nil {
+				// log error to stderr
+				log.Printf("gost/TaskPool.run error: %s", err.Error())
+			}
+		},
+		// catch handler
+		func(rec interface{}) {
+			p.safeRun(workerID, q)
```

Comment:
not exec again when panic, in my opinion

Reply:
it only start a new goroutine to consume the q, failed job won't repeat.

Reply:
call safeRun again in catch handler will exec again， or I misunderstand？

Reply:
the failed job won't exec agian. 
the method `safeRun` don't run job immediately, it only bind a queue to a goroutine.

Reply:
@AlexStocks dose I understand correctly ? 😄

Reply:
IMP，catch handler is not needed

Reply:
> IMP，catch handler is not needed

@micln what mr @wongoo said is right. If u use it in this way, it is like to gxruntime. GoUnterminated.

Reply:
removed